$OpenBSD: patch-xvsmooth_c,v 1.1 2011/05/25 09:09:56 sthen Exp $

Fix off-by-one's resulting in dereferncing out-of-bound memory.
http://permalink.gmane.org/gmane.os.openbsd.ports/48385

--- xvsmooth.c.orig	Sun May 22 23:48:28 2011
+++ xvsmooth.c	Sun May 22 23:52:31 2011
@@ -369,7 +369,7 @@ int   is24, swide, shigh, dwide, dhigh;
 
   lastline = linecnt = 0;
 
-  for (i=0, clptr=pic824; i<=shigh; i++, clptr+=swide*bperpix) {
+  for (i=0, clptr=pic824; i<shigh; i++, clptr+=swide*bperpix) {
     ProgressMeter(0, shigh, i, "Smooth");
     if ((i&15) == 0) WaitCursor();
 
@@ -468,7 +468,7 @@ int   is24, swide, shigh, dwide, dhigh;
   lastline = linecnt = pixR = pixG = pixB = 0;
   cptr = pic824;
 
-  for (i=0; i<=shigh; i++) {
+  for (i=0; i<shigh; i++) {
     ProgressMeter(0, shigh, i, "Smooth");
     if ((i&15) == 0) WaitCursor();
 
